### PR TITLE
Plot fixes and improvements

### PR DESCRIPTION
**Pull request by crvernon**
This pull request introduces comprehensive testing for the `fit_logit` and `plot_contour_map` utility functions, as well as minor improvements and clarifications in their implementation. The main focus is on increasing test coverage, improving robustness, and cleaning up the code for clarity and maintainability.

### Testing enhancements

* Added a new test suite `msdbook/tests/test_utils.py` with extensive tests for `fit_logit` and `plot_contour_map`, covering normal operation, edge cases (empty data, invalid predictors), and logistic regression with interaction terms. This includes parameterized tests and checks for expected model properties and output values.

### Code improvements and fixes

* Updated `fit_logit` in `msdbook/utils.py` to remove the deprecated `disp` argument and explicitly specify the fitting method (`method='bfgs'`) for compatibility with current `statsmodels` versions. Also added clarifying comments and improved variable naming.
* Improved the `plot_contour_map` function by clarifying comments, ignoring unnecessary warnings, and ensuring scatter points are plotted based on the data. [[1]](diffhunk://#diff-6297400240ddad776575a5c2819117ab1c4e6c4dbad0be21f95c2ce678acff90R39-R44) [[2]](diffhunk://#diff-6297400240ddad776575a5c2819117ab1c4e6c4dbad0be21f95c2ce678acff90L6-R29)

### Minor formatting improvements

* Reformatted meta tags and script blocks in `landing/src/app.html` for better readability and maintainability, and added a new analytics script.

### CODE REVIEW

File: msdbook/utils.py [hunk 1]
```diff
@@ -3,30 +3,30 @@
 import numpy as np
 import statsmodels.api as sm
 
-
 def fit_logit(dta, predictors):
     """Logistic regression"""
 
-    # concatenate intercept column of 1s
+    # Add intercept column of 1s
     dta["Intercept"] = np.ones(np.shape(dta)[0])
```
[the remark this answers is not in the document]
Comment (crvernon) [reply msdbook/utils.py:31]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/IMMM-SFA/msd_uncertainty_ebook/pull/159#discussion_r2535507963)

File: msdbook/utils.py [hunk 2]
```diff
@@ -3,30 +3,30 @@
...
 
-
 def plot_contour_map(
     ax, result, dta, contour_cmap, dot_cmap, levels, xgrid, ygrid, xvar, yvar, base
 ):
     """Plot the contour map"""
 
-    # TODO:  see why this warning is being raised about the tight layout
+    # Ignore tight layout warnings
     warnings.filterwarnings("ignore")
```
[the remark this answers is not in the document]
Comment (crvernon) [reply msdbook/utils.py:47]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/IMMM-SFA/msd_uncertainty_ebook/pull/159#discussion_r2535507976)

File: msdbook/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        'Predictor2': np.random.randn(n),  # Random values for Predictor2
+        'Interaction': np.random.randn(n)  # Random values for Interaction term
+    })
+
+    return df
+predictor1 = 'Predictor1'
+predictor2 = 'Predictor2'
+interaction = 'Interaction'
+intercept = 'Intercept'
+success = 'Success'
```
[the remark this answers is not in the document]
Comment (crvernon) [reply msdbook/tests/test_utils.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/IMMM-SFA/msd_uncertainty_ebook/pull/159#discussion_r2535507995)

File: msdbook/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        success: [1.0, 0.0, 1.0],
+    }), 0.0, 2.0, -2.4002923915238235e-06),
+
+    (pd.DataFrame({
+        predictor1: [0.5, 1.5, 2.5],
+        predictor2: [1.0, 2.0, 3.0],
+        interaction: [0.2, 0.4, 0.6],
+        intercept: [1.0, 1.0, 1.0],
+        success: [0.0, 1.0, 1.0],
+    }), 0.0, 2.0, -1.7925479970021486e-05)
```
[the remark this answers is not in the document]
Comment (crvernon) [reply msdbook/tests/test_utils.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/IMMM-SFA/msd_uncertainty_ebook/pull/159#discussion_r2535508077)

File: msdbook/utils.py [hunk 5]
```diff
@@ -3,30 +3,30 @@
...
 def fit_logit(dta, predictors):
     """Logistic regression"""
 
-    # concatenate intercept column of 1s
+    # Add intercept column of 1s
     dta["Intercept"] = np.ones(np.shape(dta)[0])
-    # get columns of predictors
+    
+    # Get columns of predictors
     cols = dta.columns.tolist()[-1:] + predictors + ["Interaction"]
```
[the remark this answers is not in the document]
Comment (crvernon) [reply msdbook/utils.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/IMMM-SFA/msd_uncertainty_ebook/pull/159#discussion_r2535508124)

**Comment by crvernon** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/IMMM-SFA/msd_uncertainty_ebook/pull/159#pullrequestreview-3474547527)
